core/src/main/scala/ai/lum/odinson/OdinsonIndexWriter.scala

```diff
@@ -81,9 +101,16 @@ class OdinsonIndexWriter(
     addDocuments(block)
   }
 
-  def commit(): Unit = writer.commit()
+  def commit(): Unit = {
+    writer.flush()
+    writer.commit()
+  }
 
   def close(): Unit = {
+    if (directory.listAll().contains(VOCABULARY_FILENAME)) directory.deleteFile(VOCABULARY_FILENAME)
```

Comment:
can you please add a small comment here to help ppl know why these files are being deleted? thanks!